client/src/App.js

```diff
@@ -1,13 +1,43 @@
 import "./App.css";
+import React, { useState } from "react";
+
+import UploadComponent from "./components/UploadComponent";
+import Search from "./components/Search";
+import Container from "./components/Container";
+import Footer from "./components/Footer";
+
+const App = () => {
+  const [data, setData] = useState([]);
+  const [videos, setVideos] = useState([]);
+  const [searchQuery, setSearchQuery] = useState("");
+
+  const addLocally = (newData) => {
+    data.push(newData);
+  };
+
+  const searchForVideo = () => {
+    const filteredVideos = data.filter((video) => {
+      return video.title.toLowerCase().includes(searchQuery);
+    });
+    setVideos(filteredVideos);
```

Comment:
I would generally expect you not to discard the original list of videos, because ideally you wouldn't need to re-fetch them if I remove/change my search query... Can you think how to avoid throwing away the full list of videos when searching?